src/EditorFeatures/Test2/ReferenceHighlighting/AbstractReferenceHighlightingTests.vb

```diff
@@ -55,14 +56,21 @@ Namespace Microsoft.CodeAnalysis.Editor.UnitTests.ReferenceHighlighting
                 Dim expectedTags As New List(Of String)
 
                 For Each hostDocument In workspace.Documents
-                    For Each nameAndSpans In hostDocument.AnnotatedSpans.OrderBy(Function(x) x.Value.First().Start)
-                        For Each span In nameAndSpans.Value
-                            expectedTags.Add(nameAndSpans.Key + ":" + span.ToString())
-                        Next
+                    Dim nameAndSpansList = hostDocument.AnnotatedSpans.SelectMany(
+                        Function(name) name.Value,
+                        Function(name, spans) New With {name, spans}) _
+                        .Select(Function(nameAndSpans) _
+                                    New With {.Name = nameAndSpans.name.Key,
+                                              .Span = nameAndSpans.spans
+                                    })
```

Comment:
There's a cleanup opportunity here: you could do the .Name and .Span assignments in the earlier "New With" -- otherwise we're making one set of types just to project onto a second set of types. It's a bit confusing especially because "spans" is just a single span.

Reply:
yeah, i ahve to admit i don't really understand what this code is doing.  perhaps a helper method that clearly defines what the shape of the result type is would be helpful?

Reply:
@CyrusNajmabadi @jasonmalinowski simplified the logic if either of you can take a look.